core/src/main/java/com/graphhopper/routing/util/BikeCommonFlagEncoder.java

```diff
@@ -550,7 +564,8 @@ void collect( OSMWay way, double wayTypeSpeed, TreeMap<Double, Integer> weightTo
     {
         String service = way.getTag("service");
         String highway = way.getTag("highway");
-        if (way.hasTag("bicycle", "designated") || way.hasTag("bicycle", "official"))
+        if ( ((highway!=null) && !highway.equals("steps")) && 
+              (way.hasTag("bicycle", "designated") || way.hasTag("bicycle", "official")) )
```

Comment:
>it is designated for bike then we do not need this steps check IMO? If it is tagged as steps & designated this seems more like a tagging bug

Exactly. And therefore I now do not push the priority. 

> Also why is this related to segregated?

It is true, the `step` change is unrelated to `segregated`. But I would like to keep this.